lib/puppet/transaction/report.rb

```diff
@@ -108,6 +108,12 @@ class Puppet::Transaction::Report
   #
   attr_reader :report_format
 
+  # A hash of selected puppet run settings:
+  #   :noop [Boolean] - whether the puppet run was started in noop mode
+  # @return [Hash<{String => Object}>]
+  #
+  attr_reader :settings
```

Comment:
The concept of having a hash in Puppet is an interesting one, I like it in a way. Having said that, be aware that hash traversal of this nature isn't easy in PDB yet, so we may be left just mapping this to the top level for the moment in our terminus: https://tickets.puppetlabs.com/browse/PDB-2633